subsys/bluetooth/mesh/provisioner.c

```diff
@@ -874,6 +875,54 @@ int bt_mesh_pb_remote_open_node(struct bt_mesh_rpr_cli *cli,
 		return -ENOENT;
 	}
 
+	/* Remote Provision done on same node, skip open link and send provision
+	 * message, just execute provisioning on it self.
+	 */
+	if (IS_ENABLED(CONFIG_BT_MESH_RPR_SRV) && bt_mesh_has_addr(srv->addr)) {
+		if (atomic_test_and_set_bit(bt_mesh_prov_link.flags, LINK_ACTIVE)) {
+			return -EBUSY;
+		}
+
+		LOG_DBG("net_idx %u iv_index 0x%08x, addr 0x%04x",
+			prov_device.node->net_idx, bt_mesh_cdb.iv_index, addr);
+
+		atomic_set_bit(bt_mesh_prov_link.flags, REPROVISION);
+		atomic_set_bit(bt_mesh_prov_link.flags, PROVISIONER);
+		bt_mesh_prov_link.addr = addr;
+		bt_mesh_prov_link.bearer = &pb_remote_cli;
+		bt_mesh_prov_link.role = &role_provisioner;
+		prov_device.net_idx = prov_device.node->net_idx;
+		prov_device.attention_duration = 0;
+
+		const uint8_t *key = bt_mesh_pub_key_get();
+
+		if (!key) {
+			LOG_ERR("No public key available");
+			return -ENOEXEC;
+		}
+
+		if (bt_mesh_dhkey_gen(key, NULL, bt_mesh_prov_link.dhkey)) {
+			LOG_ERR("Failed to generate DHKey");
+			return -ENOEXEC;
+		}
+		LOG_DBG("DHkey: %s", bt_hex(bt_mesh_prov_link.dhkey, DH_KEY_SIZE));
+
+		err = bt_mesh_dev_key(bt_mesh_prov_link.dhkey,
+				      bt_mesh_prov_link.prov_salt, prov_device.new_dev_key);
+		if (err) {
+			LOG_ERR("Unable to generate device key");
+			return err;
+		}
+
+		bt_mesh_dev_key_cand(prov_device.new_dev_key);
+		bt_mesh_reprovision(addr);
+		bt_mesh_dev_key_cand_activate();
+		/* Mark the link that was never opened as closed. */
+		atomic_set_bit(bt_mesh_prov_link.flags, COMPLETE);
+		prov_link_closed(PROV_BEARER_LINK_STATUS_SUCCESS);
```

Comment:
Would it make sense to call `reprovision_complete` from `prov_device.c` here? Or we at least should call `reprovisioned` callback as we call it in `prov_device.c`.